### DISCUSSION TITLE
Manage language from url

### DISCUSSION THREAD

**Discussion by hasanaktas** [expo/router#545] [Q&A]
hello, as in next.js, how can I follow a way to add language support in URLs. I need such a structure as I want to use the same urls as my website.

default language:en
test.com => en
test.com/en =>en
test.com/de =>de

test.com/products/1=>en
test.com/en/products/1=>en

test.com/de/products/1=>de

**Comment by wcastand** [comment]
i'd say you would need to create in `app` folder a folder with `[lang]` as name 

which should be transformed into dynamic params that will appear in the when using the hook
https://docs.expo.dev/router/reference/search-parameters/

i'm not 100% sure dynamic folders are a thing tho.

a good place to find stuff is the examples. https://github.com/expo/router/tree/main/apps/sandbox/app_15

dynamic folder looks doable :)
